### PR TITLE
fix: change event not called when item selected via click

### LINKED ISSUE #792: on-change callback not working on version 4.2.6.
### Description
I have a combo box with an `on-change` event handler. The event handler only gets called when I clear the selection, not changing the selection. It happens in version 4.2.6 when using in Polymer 2.
```html
<vaadin-combo-box
  theme="small"
  hidden="[[_hideEnvironmentSwitch]]"
  label="Change Environment"
  on-change="_changeEnvironment"
  items="[[_environments]]"
  value="{{_selectedEnvironment}}">
</vaadin-combo-box>
```

```js
_changeEnvironment() {
  debugger
  if (this._selectedEnvironment) {
    // some logic
  }
  this._toggleEnvironmentSelector();
}
```

### PR DESCRIPTION
**Pull request by Haprog**
Fixes #792

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-combo-box/796)

### CODE REVIEW

File: src/vaadin-combo-box-mixin.html [hunk 1]
```diff
@@ -872,6 +872,7 @@
 
       if (this.selectedItem !== e.detail.item) {
         this.selectedItem = e.detail.item;
+        this._detectAndDispatchChange();
```
Comment (web-padawan) [inline src/vaadin-combo-box-mixin.html]: adding this line here means that the following one has stopped working:
https://github.com/vaadin/vaadin-combo-box/blob/f3d2c44f65580e34a2b4c4087c95eb555f47484f/src/vaadin-combo-box-mixin.html#L599

Can we now remove that one and see if all the tests still pass?

Comment (Haprog) [reply src/vaadin-combo-box-mixin.html]: No, that one should still work for other cases, but here the problem (regression cause) was that now `close()` is called before the `selectedItem` is set so there are no changes at that point. See:
```js
      if (this.opened) {
        this.close();
      }
      if (this.selectedItem !== e.detail.item) {
        this.selectedItem = e.detail.item;
        this._detectAndDispatchChange();
      }
```

Comment (Haprog) [reply src/vaadin-combo-box-mixin.html]: But I will check if that other one could maybe be not needed anymore. I'll try running the tests without it, but I think it's still needed for when the overlay is closing is triggered in other ways.

Comment (Haprog) [reply src/vaadin-combo-box-mixin.html]: If that other `this._detectAndDispatchChange();` is removed then 3 tests fail.
image

Comment (web-padawan) [reply src/vaadin-combo-box-mixin.html]: Ok, that's what I suspected but needed to clarify that.

**Comment by Haprog** [comment]
This fix will be part of the next patch release (presumably v4.2.7) which will probably be released today.

**Comment by Haprog** [comment]
The fix is now in [v4.2.7](https://github.com/vaadin/vaadin-combo-box/releases/tag/v4.2.7).

**Pull request merged**